components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -839,7 +840,55 @@ void AdsServiceImpl::DetectUncertainFuture(const uint32_t number_of_start) {
 void AdsServiceImpl::OnDetectUncertainFuture(const uint32_t number_of_start,
                                              const bool is_uncertain_future) {
   ads::mojom::SysInfoPtr sys_info = ads::mojom::SysInfo::New();
+
+  // TODO(https://github.com/brave/brave-browser/issues/13793): Transition ads
+  // to components which will then provide access to |kFeatureName| and
+  // command-line switches rather than using hard coded strings below.
+  const auto* command_line = base::CommandLine::ForCurrentProcess();
+  if (command_line->HasSwitch("variations-server-url") ||
+      command_line->HasSwitch("variations-insecure-server-url") ||
+      command_line->HasSwitch("fake-variations-channel") ||
+      command_line->HasSwitch("variations-override-country")) {
+    sys_info->did_override_command_line_args_flag = true;
+  } else {
+    std::string switch_value;
```

Comment:
Could we collect switch values into base::flat_set?

Reply:
Will fix this as part of decoupling

Reply:
Fixed